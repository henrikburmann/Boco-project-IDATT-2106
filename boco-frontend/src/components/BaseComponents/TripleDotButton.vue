<template>
  <button
    id="dropdownDefault"
    data-dropdown-toggle="dropdown"
    class="w-10 h-10 text-gray-500 dark:text-gray-400 hover:bg-gray-100 dark:hover:bg-gray-700 focus:outline-none focus:ring-4 focus:ring-gray-200 dark:focus:ring-gray-700 rounded-lg float-right text-sm p-1.5"
    type="button"
  >
    <DotsVerticalIcon class="w-6 h-6" />
  </button>
</template>
<script>
import { DotsVerticalIcon } from "@heroicons/vue/outline";

/**
 * Triple dot button
 */
export default {
  name: "TripleDotButton",
  components: {
    DotsVerticalIcon,
  },
};
</script>
