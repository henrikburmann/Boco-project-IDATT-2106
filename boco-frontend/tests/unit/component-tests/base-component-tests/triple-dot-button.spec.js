import { mount } from "@vue/test-utils";
import TripleDotButton from "@/components/BaseComponents/TripleDotButton.vue";

describe("TripleDotButton component", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mount(TripleDotButton);
  });

  it("renders correctly", () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it("is instantiated", () => {
    expect(wrapper.exists()).toBeTruthy();
  });
});
