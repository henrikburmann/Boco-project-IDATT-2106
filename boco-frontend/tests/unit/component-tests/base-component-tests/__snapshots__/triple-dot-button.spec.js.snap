// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TripleDotButton component renders correctly 1`] = `
<button
  class="w-10 h-10 text-gray-500 dark:text-gray-400 hover:bg-gray-100 dark:hover:bg-gray-700 focus:outline-none focus:ring-4 focus:ring-gray-200 dark:focus:ring-gray-700 rounded-lg float-right text-sm p-1.5"
  data-dropdown-toggle="dropdown"
  id="dropdownDefault"
  type="button"
>
  <svg
    aria-hidden="true"
    class="w-6 h-6"
    fill="none"
    stroke="currentColor"
    stroke-width="2"
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M12 5v.01M12 12v.01M12 19v.01M12 6a1 1 0 110-2 1 1 0 010 2zm0 7a1 1 0 110-2 1 1 0 010 2zm0 7a1 1 0 110-2 1 1 0 010 2z"
      stroke-linecap="round"
      stroke-linejoin="round"
    />
  </svg>
</button>
`;
